nighthawk/source/client/client_worker_impl.cc

```diff
@@ -0,0 +1,69 @@
+#include "nighthawk/source/client/client_worker_impl.h"
+
+#include "nighthawk/source/common/platform_util_impl.h"
+
+namespace Nighthawk {
+namespace Client {
+
+ClientWorkerImpl::ClientWorkerImpl(Envoy::Api::Api& api, Envoy::ThreadLocal::Instance& tls,
+                                   const BenchmarkClientFactory& benchmark_client_factory,
+                                   const SequencerFactory& sequencer_factory, const Uri uri,
+                                   Envoy::Stats::StorePtr&& store, const int worker_number,
+                                   const Envoy::MonotonicTime starting_time)
+    : WorkerImpl(api, tls, std::move(store)), uri_(uri), worker_number_(worker_number),
+      starting_time_(starting_time),
+      benchmark_client_(benchmark_client_factory.create(api, *dispatcher_, *store_, uri)),
+      sequencer_(sequencer_factory.create(time_source_, *dispatcher_, *benchmark_client_)) {}
+
+void ClientWorkerImpl::simpleWarmup() {
+  ENVOY_LOG(debug, "> worker {}: warming up.", worker_number_);
+  // TODO(oschaaf): Maybe add BenchmarkClient::warmup() and call that here.
+  // Ideally that would warm up the pool better, by prefetching the requested amount of
+  // connections. Currently it is possible to use less connections then specified if
+  // completions are fast enough. While this may be an assert, it may also be annoying
+  // when comparing results to some other tools, which do open up the specified amount
+  // of connections.
+  benchmark_client_->tryStartOne([this] { dispatcher_->exit(); });
+  dispatcher_->run(Envoy::Event::Dispatcher::RunType::Block);
+}
+
+void ClientWorkerImpl::delayStart() {
+  PlatformUtilImpl platform_util;
+  // TODO(oschaaf): We could use dispatcher to sleep, but currently it has a 1 ms resolution
+  // which is rather coarse for our purpose here.
```

Comment:
How come this is too coarse? I would have though the initial delay would be O(seconds), so that would be fine. Would make the code a bit cleaner.